### PR TITLE
Fix error loading older sender data from storage

### PR DESCRIPTION
**Pull request by andybalaam**
Fixes https://github.com/matrix-org/matrix-rust-sdk/issues/4424

### CODE REVIEW

File: crates/matrix-sdk-crypto/src/olm/group_sessions/sender_data.rs [hunk 1]
```diff
@@ -215,6 +215,7 @@ enum SenderDataReader {
         legacy_session: bool,
     },
 
+    #[serde(alias = "SenderUnverifiedButPreviouslyVerified")]
     VerificationViolation(KnownSenderData),
```
Comment (richvdh) [inline crates/matrix-sdk-crypto/src/olm/group_sessions/sender_data.rs:219]: There are a whole bunch of other renames in #4067 that we need to check for; notably `OwnUserIdentityVerifiedState` (https://github.com/matrix-org/matrix-rust-sdk/pull/4067/files#diff-d28a9285dd9fd0d1ced29b1986dc92d7a754bb846a24cab48ec78073a1581166R874) which was the one that caused problems for Valere, but also `VerificationLevel`, and there may be others.

**Comment by poljar** [comment]
This talks about a crash, but we just changed the serialization format, deserialization failures should not result in crashes. So we need to also find where the crash happens and convert that to an error.

**Comment by andybalaam** [comment]
> This talks about a crash, but we just changed the serialization format, deserialization failures should not result in crashes. So we need to also find where the crash happens and convert that to an error.

It was a crash, right @BillCarsonFr ? Can you paste the stack trace?

File: crates/matrix-sdk-crypto/src/olm/group_sessions/sender_data.rs (hunk 1 above)
Comment (andybalaam) [reply crates/matrix-sdk-crypto/src/olm/group_sessions/sender_data.rs:219]: Sorry, was rushing to find a fix and missed the other places. Will expand this.

Comment (andybalaam) [reply crates/matrix-sdk-crypto/src/olm/group_sessions/sender_data.rs:219]: Added commits dealing with the other cases

**Review by richvdh — APPROVED** [review]
LGTM

**Comment by andybalaam** [comment]
> Alright, but that's EA not catching an exception, right? The SDK itself doesn't crash.

Correct. I fixed the PR description and will fix the commits in my rebase.

File: crates/matrix-sdk-common/src/deserialized_responses.rs [hunk 2]
```diff
@@ -179,6 +179,7 @@ pub enum VerificationLevel {
 
     /// The message was sent by a user identity we have not verified, but the
     /// user was previously verified.
+    #[serde(alias = "PreviouslyVerified")]
```
[the remark this answers is not in the document]
Comment (andybalaam) [reply crates/matrix-sdk-common/src/deserialized_responses.rs:182]: I feel like it's quite obvious and the tests serve to document it.

**Review by poljar — APPROVED** [review]

**Comment by andybalaam** [comment]
Pausing this PR so we can make a release branch to release this for crypto-wasm based on the version used in matrix-rust-sdk-crypto-wasm 11.0 (70bcddfba5e19f2d0fd11577edb55b9868f89678)

**Comment by poljar** [comment]
Err, please don't pause it? The WASM bindings are not the only ones affected by this.

**Comment by andybalaam** [comment]
I merged the new PR into `release-for-crypto-wasm-11`, then in merged that branch into main in commit 03947618ffd0eb3e603536e6c9575077f6a91a37

**Pull request closed without merging**